unity/Assets/Scripts/AgentManager.cs

```diff
@@ -346,68 +344,190 @@ public BaseFPSAgentController ReturnPrimaryAgent()
         }
     }
 
-	public void AddThirdPartyCamera(ServerAction action) {
-		GameObject gameObject = new GameObject("ThirdPartyCamera" + thirdPartyCameras.Count);
-		gameObject.AddComponent(typeof(Camera));
-		Camera camera = gameObject.GetComponentInChildren<Camera>();
+    private void updateCameraProperties(
+        Camera camera,
+        Vector3 position,
+        Vector3 rotation,
+        float fieldOfView,
+        string skyboxColor,
+        bool? orthographic,
+        float? orthographicSize,
+        string actionName
+    ) {
+        // update the position and rotation
+        camera.gameObject.transform.position = position;
+        camera.gameObject.transform.eulerAngles = rotation;
+
+        // updates the camera's perspective
+        camera.fieldOfView = fieldOfView;
+        if (orthographic != null) {
+            camera.orthographic = (bool) orthographic;
+            if (orthographic == true && orthographicSize != null) {
+                camera.orthographicSize = (float) orthographicSize;
+            }
+        }
 
-		camera.cullingMask = ~(1 << 11);
+        // supports a solid color skybox, which work well with videos and images (i.e., white/black/orange/blue backgrounds)
+        bool actionFailed = false;
+        if (skyboxColor == "default") {
+            camera.clearFlags = CameraClearFlags.Skybox;
+        } else if (skyboxColor != null) {
+            Color color;
+            bool successfullyParsed = ColorUtility.TryParseHtmlString(skyboxColor, out color);
```

Comment:
This is one of the cooler functions I've come across, which allows for color parsing from ANY HTML string color, like:
```css
rgb(0, 255, 255)
hsl(180, 50%, 50%)
#FAFAFA
#AAA
orange
black
green
blue
```